packages/types/src/element/textElement.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import type { Element } from './element';
+import type { Pattern } from './pattern';
+
+export type FontStyle = 'normal' | 'italic';
+export enum FontVariantStyle {
+  Normal = 0,
+  Italic = 1,
+}
+
+export type FontWeight = 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;
+
+export type FontVariant = [FontVariantStyle, FontWeight];
+
+export interface Font {
+  family: string;
+  service?: string;
+  weights?: FontWeight[];
+  styles?: FontStyle[];
+  variants?: FontVariant[];
+  fallbacks?: string[];
```

Comment:
`metrics` seems to be missing from here.

Reply:
(it's actually quite the mess that we put the metrics in the story data at all, they definitely don't belong there, and even if they did, they should be summarized somewhere on the top level, not duplicated per text element)

Reply:
True, it was discussed previously as well, perhaps @swissspidy remembers why all the font information was saved per element even though it's the same for the same font.

Reply:
> (it's actually quite the mess that we put the metrics in the story data at all, they definitely don't belong there, and even if they did, they should be summarized somewhere on the top level, not duplicated per text element)

Yes, absolutely. Which is why I created #12261 to move them to the top level. It would definitely help make things more clean.

I think originally we stored it per-element because it had the advantage of having all the information right away when dealing with text elements, without having to do round trips to the story provider to get font information and pass that along everywhere. Just at the cost of a little redundancy. I don't think we had metrics yet back then.
But yeah especially with the introduction of metrics (and having to keep metrics updated everywhere) this is now a bit too much and the top level storage would be cleaner.

Reply:
And if you need a type:

```ts
interface FontMetrics {
  upm: number;
  asc: number;
  des: number;
  tAsc: number;
  tDes: number;
  tLGap: number;
  wAsc: number;
  wDes: number;
  xH: number;
  capH: number;
  yMin: number;
  yMax: number;
  hAsc: number;
  hDes: number;
  lGap: number;
}
```